tests/acceptance/nested-preprocessor-test-slow.js

```diff
@@ -0,0 +1,332 @@
+function generatePreprocessor(context) {
+  return processTemplate(setupPreprocessorRegistryFixture.toString(), context);
+}
+
+// THIS IS A FIXTURE. It also happens to be valid JavaScript.
+var preprocessTreeFixture = function(type, tree) {
+  if (type === 'all') { return tree; }
+
+  // THIS IS A WORKAROUND FOR https://github.com/ember-cli/ember-cli/issues/6512
+  if (type === 'css' && tree.toString().indexOf('Concat: Vendor Styles') !== -1) {
+    return tree;
+  }
+
+  var stew = require('broccoli-stew');
+  var addon = this;
+
+  // We're going to add a marker. The `postprocessTree` function will remove
+  // the marker if present and identify whether or not it found the marker.
+  var marker = '// ' + addon.name + '-preprocessTree(' + type + ')' + ' /';
+
+  tree = stew.map(tree, function(content, relativePath) {
+    var localMarker = marker + relativePath;
+    addon._findHost().project.ui.writeLine(localMarker);
+    return localMarker + '\n' + content;
+  });
+
+  return tree;
+};
+
+// THIS IS A FIXTURE. It also happens to be valid JavaScript.
+var postprocessTreeFixture = function(type, tree) {
+  if (type === 'all') { return tree; }
+
+  var stew = require('broccoli-stew');
+  var addon = this;
+
+  var preprocessTreeMarker = '// ' + addon.name + '-preprocessTree(' + type + ')';
+  var marker = '// ' + addon.name + '-postprocessTree(' + type + ')';
+
+  // We're going to inspect state and add an appropriate marker.
+  tree = stew.map(tree, function(content, relativePath) {
+
+    // Skip assets we know that the CSS preprocessor adds.
+    var cssName = addon._findHost().name;
+    if (type === 'css') {
+      if (relativePath === 'assets/' + cssName + '.css' || relativePath === 'assets/vendor.css') {
+        return content;
+      }
+    }
+
+    // Build up the path for where the asset was previously.
+    var preprocessPath = relativePath;
+
+    if (type === 'template') {
+      preprocessPath = preprocessPath.replace(/.js$/, '.hbs');
+    } else if (type === 'css') {
+      preprocessPath = preprocessPath.replace(/^assets/, 'app/styles');
+    }
+
+    var localPreprocessTreeMarker = preprocessTreeMarker + ' /' + preprocessPath;
+    var preprocessTreeMarkerIndex = content.indexOf(localPreprocessTreeMarker);
+    var preprocessTreeMarkerLastIndex = content.lastIndexOf(localPreprocessTreeMarker);
+
+    var localMarker = marker;
+
+    if (preprocessTreeMarkerIndex === -1) {
+      localMarker += '-no-preprocessTree';
+      localMarker += ' /' + relativePath;
+      addon._findHost().project.ui.writeLine(localMarker);
+      return localMarker + '\n' + content;
+    } else {
+      localMarker += '-removed-preprocessTree';
+      localMarker += ' /' + relativePath;
+      addon._findHost().project.ui.writeLine(localMarker);
+      return content.replace(localPreprocessTreeMarker, localMarker);
+    }
+  });
+
+  return tree;
+};
+
+describe('Acceptance: nested preprocessor tests.', function() {
+  this.timeout(1000 * 60 * 10);
```

Comment:
I'd prefer to just use the whole number and add a comment for human form, thoughts?

```js
this.timeout(600000); // 10 minutes
```